src/commands/Minion/fletch.ts

```diff
@@ -0,0 +1,139 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+
+import { BotCommand } from '../../lib/BotCommand';
+import {
+	stringMatches,
+	formatDuration,
+	rand,
+	itemNameFromID,
+	removeItemFromBank
+} from '../../lib/util';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { Time, Activity, Tasks } from '../../lib/constants';
+import { FletchingActivityTaskOptions } from '../../lib/types/minions';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import bankHasItem from '../../lib/util/bankHasItem';
+import Fletching from '../../lib/skilling/skills/fletching/fletching';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '[quantity:int{1}|name:...string] [name:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	async run(msg: KlasaMessage, [quantity, fletchName = '']: [null | number | string, string]) {
+		if (!msg.author.hasMinion) {
+			throw `You dont have a minion`;
+		}
+
+		if (msg.flagArgs.items) {
+			return msg.channel.sendFile(
+				Buffer.from(
+					Fletching.Fletchables.map(
+						item =>
+							`${item.name} - lvl ${item.level} : ${Object.entries(item.inputItems)
+								.map(entry => `${entry[1]} ${itemNameFromID(parseInt(entry[0]))}`)
+								.join(', ')}`
+					).join('\n')
+				),
+				`Available fletching items.txt`
+			);
+		}
+
+		if (msg.author.minionIsBusy) {
+			return msg.send(msg.author.minionStatus);
+		}
+
+		if (typeof quantity === 'string') {
+			fletchName = quantity;
+			quantity = null;
+		}
+
+		const Fletch = Fletching.Fletchables.find(item => stringMatches(item.name, fletchName));
```

Comment:
this variable should be called `fletchableItem` or `itemToFletch` or something like that.